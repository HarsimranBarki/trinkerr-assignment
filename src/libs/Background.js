const Background = ({ children }) => {
  return (
    // Remove transition-all to disable the background color transition.
    <div className="bg-white dark:bg-gray-800 dark:text-gray-100 h-screen transition-all">
      {children}
    </div>
  );
};
export default Background;
