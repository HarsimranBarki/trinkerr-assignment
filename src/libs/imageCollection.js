import one from "../images/one.png";
import two from "../images/two.png";
import three from "../images/three.png";
import four from "../images/four.png";
import five from "../images/five.png";

export const imageCollection = [
  {
    id: 1,
    name: "one",
    imageUrl: one,
    swiped: false,
  },
  {
    id: 2,
    name: "two",
    imageUrl: two,
    swiped: false,
  },
  {
    id: 3,
    name: "three",
    imageUrl: three,
    swiped: false,
  },
  {
    id: 4,
    name: "four",
    imageUrl: four,
    swiped: false,
  },
  {
    id: 5,
    name: "five",
    imageUrl: five,
    swiped: false,
  },
];
