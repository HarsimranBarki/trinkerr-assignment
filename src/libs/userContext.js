import React, { useMemo, useState } from "react";

export const UserContext = React.createContext(null);

export const UserProvider = ({ children }) => {
  const [user, setUser] = useState(
    JSON.parse(localStorage.getItem("user")) || null
  );
  const providerValue = useMemo(() => ({ user, setUser }), [user, setUser]);

  return (
    <UserContext.Provider value={providerValue}>
      {children}
    </UserContext.Provider>
  );
};
