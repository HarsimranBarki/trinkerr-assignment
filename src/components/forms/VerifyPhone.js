import { motion } from "framer-motion";
import React, { useContext, useState } from "react";
import toast, { Toaster } from "react-hot-toast";
import { UserContext } from "../../libs/userContext";
import { FaChevronCircleRight } from "react-icons/fa";

const VerifyPhone = ({ setSignupState }) => {
  const [phone, setPhone] = useState("");
  const toastGenerate = (message) => toast.error(message);
  const { user, setUser } = useContext(UserContext);

  // Regex For Formatting Input Should Be Number
  const onlyNumbers = (e) => {
    e.preventDefault();
    let value = e.target.value.replace(/[^\d]/g, "");
    setPhone(value);
  };

  const validateInput = (e) => {
    e.preventDefault();
    if (phone.length !== 10) return toastGenerate("Number should be 10 Digits");
    setUser({ ...user, phone: phone });
    localStorage.setItem("user", JSON.stringify({ phone: phone }));
    setSignupState("otp");
  };

  return (
    <motion.form
      className="mt-10 "
      onSubmit={(e) => validateInput(e)}
      initial={{ opacity: 0, x: -10 }}
      animate={{ opacity: 1, x: 0 }}
      exit={{ opacity: 0, x: 10 }}
    >
      <Toaster position="bottom-center" reverseOrder={false} />
      <input
        type="text"
        className="form-input -1 px-3 py-2 md:px-4 md:py-2 border-gray-300 rounded mt-2 shadow-inner"
        placeholder="Enter Your Number"
        value={phone}
        data-testid="input"
        onChange={(e) => onlyNumbers(e, "phone")}
        id="phoneInput"
        autoFocus
      />
      <br />
      <button
        type="submit"
        id="verifyPhoneButton"
        data-testid="button"
        className="bg-blue-600 text-indigo-50 px-3 py-2 md:px-5 md:py-2  font-medium rounded mt-5 hover:bg-blue-700 transition flex items-center"
      >
        Get OTP <FaChevronCircleRight className="inline ml-2" />
      </button>
    </motion.form>
  );
};

export default VerifyPhone;
