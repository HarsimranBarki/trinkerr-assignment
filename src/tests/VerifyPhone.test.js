import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import VerifyPhone from "../components/forms/VerifyPhone";

import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { UserProvider } from "../libs/userContext";
import { render, screen, fireEvent } from "@testing-library/react";

Enzyme.configure({ adapter: new Adapter() });

describe("VerifyPhone", () => {});
it("VerifyPhone: should render correctly", () => {
  const tree = renderer
    .create(
      mount(
        <UserProvider>
          <VerifyPhone />
        </UserProvider>
      )
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

// describe("test phone input", () => {
//   it("Updates on change", () => {
//     render(
//       <UserProvider>
//         <VerifyPhone />
//       </UserProvider>
//     );

//     const buttonPhone = screen.queryByTestId("button");
//     const inputPhone = screen.queryByTestId("input");

//     fireEvent.change(inputPhone, { target: { value: "1234567890" } });
//     // buttonPhone.simulate("click");
//     expect(inputPhone.value).toBe("1234567890");
//     let vals = inputPhone.value.length;
//     expect(vals).not.toBeNull();
//     expect(vals).not.toBeGreaterThan(10);
//     expect(vals).not.toBeLessThan(10);
//   });
// });

// describe("test phone input empty value", () => {
//   it("Updates on change", () => {
//     render(
//       <UserProvider>
//         <VerifyPhone />
//       </UserProvider>
//     );

//     const buttonPhone = screen.queryByTestId("button");
//     const inputPhone = screen.queryByTestId("input");

//     inputPhone.value = "";
//     fireEvent.click(buttonPhone);
//     let vals = inputPhone.value.length;
//     // buttonPhone.simulate("click");
//     expect(vals).not.toBeNull();
//   });
// });
