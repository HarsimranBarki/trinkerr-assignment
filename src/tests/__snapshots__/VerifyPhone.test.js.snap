// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyPhone: should render correctly 1`] = `
<form
  className="mt-10 "
  onSubmit={[Function]}
  style={
    Object {
      "opacity": 0,
      "transform": "translateX(-10px) translateZ(0)",
    }
  }
>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "bottom": 16,
        "left": 16,
        "pointerEvents": "none",
        "position": "fixed",
        "right": 16,
        "top": 16,
        "zIndex": 9999,
      }
    }
  />
  <input
    autoFocus={true}
    className="form-input -1 px-3 py-2 w-full md:px-4 md:py-2 border-gray-300 rounded mt-2 shadow-inner"
    data-testid="input"
    id="phoneInput"
    onChange={[Function]}
    placeholder="Enter Your Number"
    type="text"
    value=""
  />
  <br />
  <button
    className="bg-blue-600 text-indigo-50  px-3 py-2 md:px-5 md:py-2  font-medium rounded mt-5 hover:bg-blue-700 transition flex items-center"
    data-testid="button"
    id="verifyPhoneButton"
    type="submit"
  >
    Get OTP 
    <svg
      className="inline ml-2"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zm113.9 231L234.4 103.5c-9.4-9.4-24.6-9.4-33.9 0l-17 17c-9.4 9.4-9.4 24.6 0 33.9L285.1 256 183.5 357.6c-9.4 9.4-9.4 24.6 0 33.9l17 17c9.4 9.4 24.6 9.4 33.9 0L369.9 273c9.4-9.4 9.4-24.6 0-34z"
      />
    </svg>
  </button>
</form>
`;
